File: Assignment 1/Selectionsort/selection.py
import random
import sys
from time import time


def SelectionSort(arr):
    temp = 0
    arr2 = arr.copy()
    for i in range(len(arr2)):
        minimo = i
        for j in range(i+1,len(arr2)):
            if arr2[minimo] > arr2[j]:
                minimo = j
        temp = arr2[i]
        arr2[i] = arr2[minimo]
        arr2[minimo] = temp


#input de datos
archivo = open("input.txt").readlines()
print(archivo)
n = int(archivo[0])
datos = map(int,archivo[1].split())
arr = list(datos)

#medir tiempos
start = time()
SelectionSort(arr)
finish = time()

#output en un archivo
with open('output.txt', 'w') as f:
    sys.stdout = f # Change the standard output to the file we created.
    print(n)
    print(finish-start)

File: Assignment 1/Selectionsort/selection.cpp
#include <iostream>

using namespace std;

void selectionSort(int *arr,int size)
{
    int temp;
    for(int i = 0 ; i < size - 1 ;  i++ )
    {
        int minimo = i;
        for ( int j = i+1 ; j < size  ;  j++ )
        {
            if( arr[minimo] > arr[j] )
                minimo = j ;
        }

        temp = arr[i];
        arr[i] = arr[minimo];
        arr[minimo] = temp;
        
    }
}

int main()
{
    int arr[] = { 10 ,9 ,8 ,7 , 6,5 ,4, 3, 2, 1};
    selectionSort(arr,10);
    for (int i = 0 ; i < 10 ; i++ )
         cout << arr[i] << " ";
    return 0;
}
